File: sdk/java/src/main/java/com/learnerview/sab/model/ApiKeyRequest.java
package com.learnerview.sab.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Request object for creating an API key.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ApiKeyRequest {
    /**
     * Label for the API key.
     */
    private String label;

    /**
     * Whether the key should have admin privileges.
     */
    private Boolean isAdmin;

    /**
     * Allowed job types for this key (if not admin).
     */
    private List<String> allowedJobTypes;
}

File: sdk/nodejs/src/models/api-key-request.ts
//Request object for creating an API key.

export interface ApiKeyRequest {
  //Label for the API key.

  label: string;
  //Producer/tenant ID.

  producer: string;
  //Whether the key has admin privileges.

  isAdmin?: boolean;
}
//Helper function to create an API key request.

export function createApiKeyRequest(
  label: string,
  producer: string,
  isAdmin: boolean = false
): ApiKeyRequest {
  return {
    label,
    producer,
    isAdmin,
  };
}
//Helper function to create a tenant API key request.

export function createTenantApiKeyRequest(label: string, producer: string): ApiKeyRequest {
  return createApiKeyRequest(label, producer, false);
}
//Helper function to create an admin API key request.

export function createAdminApiKeyRequest(label: string, producer: string): ApiKeyRequest {
  return createApiKeyRequest(label, producer, true);
}
